crystalfontz/dbus/client.py

```diff
@@ -0,0 +1,442 @@
+    """
+    Listen for key and temperature reports.
+
+    To configure which reports to receive, use 'crystalfontz keypad reporting' and
+    'crystalfontz temperature reporting' respectively.
+    """
+
+    raise NotImplementedError("listen")
+
+
+@main.command(help="0 (0x00): Ping command")
+@click.argument("payload", type=BYTES)
+@async_command
+@pass_client
+async def ping(client: DbusClient, payload: bytes) -> None:
+    pong = await client.ping(payload, TimeoutM.none, RetryTimesM.none)
+    echo(pong)
+
+
+@main.command(help="1 (0x01): Get Hardware & Firmware Version")
+@async_command
+@pass_client
+async def versions(client: DbusClient) -> None:
+    versions = await client.versions(TimeoutM.none, RetryTimesM.none)
+    echo(VersionsM.load(versions))
+
+
+@main.group(help="Interact with the User Flash Area")
+def flash() -> None:
+    pass
+
+
+@flash.command(name="write", help="2 (0x02): Write User Flash Area")
+@click.argument("data", type=BYTES)
+@async_command
+@pass_client
+async def write_user_flash_area(client: DbusClient, data: bytes) -> None:
+    await client.write_user_flash_area(data, TimeoutM.none, RetryTimesM.none)
+
+
+@flash.command(name="read", help="3 (0x03): Read User Flash Area")
+@async_command
+@pass_client
+async def read_user_flash_area(client: DbusClient) -> None:
+    flash = await client.read_user_flash_area(TimeoutM.none, RetryTimesM.none)
+    echo(flash)
+
+
+@main.command(help="4 (0x04): Store Current State as Boot State")
+@async_command
+@pass_client
+async def store(client: DbusClient) -> None:
+    await client.store_boot_state(TimeoutM.none, RetryTimesM.none)
+
+
+@main.group(help="5 (0x05): Reboot LCD, Reset Host, or Power Off Host")
+def power() -> None:
+    pass
+
+
+@power.command(help="Reboot the Crystalfontx LCD")
+@async_command
+@pass_client
+async def reboot_lcd(client: DbusClient) -> None:
+    await client.reboot_lcd(TimeoutM.none, RetryTimesM.none)
+
+
+@power.command(help="Reset the host, assuming ATX control is configured")
+@async_command
+@pass_client
+async def reset_host(client: DbusClient) -> None:
+    await client.reset_host(TimeoutM.none, RetryTimesM.none)
+
+
+@power.command(help="Turn the host's power off, assuming ATX control is configured")
+@async_command
+@pass_client
+async def shutdown_host(client: DbusClient) -> None:
+    await client.shutdown_host(TimeoutM.none, RetryTimesM.none)
+
+
+@main.command(help="6 (0x06): Clear LCD Screen")
+@async_command
+@pass_client
+async def clear(client: DbusClient) -> None:
+    await client.clear_screen(TimeoutM.none, RetryTimesM.none)
+
+
+@main.group(help="Set LCD contents for a line")
+def line() -> None:
+    pass
+
+
+@line.command(name="1", help="7 (0x07): Set LCD Contents, Line 1")
+@click.argument("line")
+@async_command
+@pass_client
+async def set_line_1(client: DbusClient, line: str) -> None:
+    await client.set_line_1(line.encode("utf-8"), TimeoutM.none, RetryTimesM.none)
```

Comment:
These commands (and the ones in the regular CLI) should accept bytes.

Reply:
Filed in #63 .

Reply:
I want to fix this prior to pulling this PR in.

Reply:
Fixed in this PR. It's a straightforward and low priority fix, and I didn't want to juggle merging - especially given the need to int test.